### PR TITLE
add notification examples for rich formatting and best practices

### PR DESCRIPTION
**Pull request by fhdhsni**
Notification examples for:
- Icons and Other Rich Formatting
- Best practices

### CODE REVIEW

File: frontend/demo/component/notification/notification-popup.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import 'Frontend/demo/init'; // hidden-source-line
+import '@vaadin/flow-frontend/contextMenuConnector.js'; // hidden-source-line
```
Comment (sissbruecker) [inline frontend/demo/component/notification/notification-popup.ts:2]: Just merged https://github.com/vaadin/docs/pull/655, which means this is not required anymore.

Comment (fhdhsni) [reply frontend/demo/component/notification/notification-popup.ts]: done

File: src/main/java/com/vaadin/demo/component/notification/NotificationRich.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+@Route("notification-rich")
+public class NotificationRich extends HorizontalLayout {
+    public static ComponentEventListener<ClickEvent<Button>> getClickHandler(
+            Consumer<Notification> notificationConsumer
+    ) {
+        return clickEvent -> {
+            Button btn = clickEvent.getSource();
+            btn.setEnabled(false);
+
+            Notification notification = new Notification();
```
[the remark this answers is not in the document]
Comment (fhdhsni) [reply src/main/java/com/vaadin/demo/component/notification/NotificationRich.java]: done

File: src/main/java/com/vaadin/demo/component/notification/NotificationRich.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+import com.vaadin.flow.router.Route;
+import com.vaadin.demo.DemoExporter; // hidden-source-line
+
+import java.util.function.Consumer;
+
+import static com.vaadin.flow.component.button.ButtonVariant.*;
+
+@Route("notification-rich")
+public class NotificationRich extends HorizontalLayout {
+    public static ComponentEventListener<ClickEvent<Button>> getClickHandler(
```
[the remark this answers is not in the document]
Comment (fhdhsni) [reply src/main/java/com/vaadin/demo/component/notification/NotificationRich.java]: done

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**